File: customer-services-api/src/main/java/com/wks/servicemarketplace/customerservice/api/AddressAddedEvent.java
package com.wks.servicemarketplace.customerservice.api;

import com.wks.servicemarketplace.common.AddressId;
import com.wks.servicemarketplace.common.CustomerId;
import com.wks.servicemarketplace.common.events.DomainEvent;
import com.wks.servicemarketplace.common.events.EventType;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.NonNull;
import lombok.Value;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Value
@AllArgsConstructor
public class AddressAddedEvent implements DomainEvent {

    private final AddressId externalId;

    @NonNull
    private final CustomerId customerExternalId;

    @NonNull
    private final String name;

    @NonNull
    private final String line1;

    private final String line2;

    @NotBlank
    private final String city;

    @NotNull
    private final String country;

    @NonNull
    private final BigDecimal latitude;

    @NonNull
    private final BigDecimal longitude;

    @NonNull
    private final String createdBy;

    @NonNull
    private final Long version;

    @Override
    public EventType getEventType() {
        return EventType.ADDRESS_ADDED;
    }

    @Override
    public String getEntityType() {
        return "Address";
    }
}

File: order-service/src/main/kotlin/com/wks/servicesmarketplace/orderservice/core/models/serviceorder/events/AddressAddedEvent.kt
package com.wks.servicesmarketplace.orderservice.core.models.serviceorder.events

import java.math.BigDecimal

data class AddressAddedEvent(
        val externalId: Long,
        val customerExternalId: Long,
        val name: String,
        val line1: String,
        val line2: String?,
        val city: String,
        val country: String,
        val latitude: BigDecimal,
        val longitude: BigDecimal,
        val createdBy: String,
        val version: Long
)